lib/Frontend/Frontend.cpp

```diff
@@ -953,6 +947,7 @@ bool CompilerInstance::parsePartialModulesAndLibraryFiles(
 
 void CompilerInstance::parseAndTypeCheckMainFileUpTo(
```

Comment:
At this point, there probably isn't anything special about the MainBufferID remaining. You might want to try removing this function altogether, and handling the main buffer uniformly with all the others.

Reply:
@slavapestov Good idea! I'll have a go at that as a follow-up.

Thanks for the review!